### ISSUE TITLE
Multi Lang Marian Translator not working (opus_mt_mul_en)

### ISSUE THREAD

**Issue by C-K-Loan** [JohnSnowLabs/spark-nlp#2472]
The model for `opus_mt_mul_en`  https://nlp.johnsnowlabs.com/2020/12/28/opus_mt_mul_en_xx.html is translating nothing . I.e. Translation field is missing in the metadata . This is the multi-language model. It does not seem to work in the original release 2.7.0 as well.  Single language translation works well

## Colab Notebook to re-produce the error
- https://colab.research.google.com/drive/1U-xFJVbaUeWEt5MTK9ntPDKtHYp79uQI?usp=sharing 

## Code to re-produce 

```
import os
! apt-get update -qq > /dev/null   
# Install java
! apt-get install -y openjdk-8-jdk-headless -qq > /dev/null
os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-8-openjdk-amd64"
os.environ["PATH"] = os.environ["JAVA_HOME"] + "/bin:" + os.environ["PATH"]
! pip install pyspark==2.4.5 spark-nlp==2.7.5
from sparknlp.annotator import *
from sparknlp.common import *
from sparknlp.base import *
from pyspark.sql import SparkSession
from pyspark.ml import Pipeline
from pyspark.sql.functions import array_contains
from pyspark.ml import Pipeline, PipelineModel
import sparknlp

from sparknlp.annotator import *
from sparknlp.pretrained import PretrainedPipeline
spark = sparknlp.start()
documentAssembler = DocumentAssembler()\
 .setInputCol("text")\
 .setOutputCol("document")

sentencerDL = SentenceDetectorDLModel.pretrained("sentence_detector_dl", "xx")\
 .setInputCols(["document"])\
  .setOutputCol("sentences")
# translate_de_en
 marian = MarianTransformer.pretrained("opus_mt_mul_en", "xx")\
 .setInputCols(["sentences"])\
 .setOutputCol("translation")

marian_pipeline = Pipeline(stages=[documentAssembler, sentencerDL, marian])
sdf = spark.createDataFrame([["Hallo wie geht es dir Ich bin hubert aus Deutschland"]]).toDF("text") 
m_fit = marian_pipeline.fit(sdf ) 
res_Df  = m_fit.transform(sdf)
res_Df.select('translation').show(truncate=False)
```

**Comment by maziyarpanahi** [comment]
Hi @C-K-Loan 

For the models that can translate from multiple source languages to another language, or one language to multiple languages, or multiple languages to multiple languages, you always need to have an identifier. (MarianMT doesn't come with a language detector, so it needs to know which language is translating to if the model is not 1-to-1 translation) 

```
a sentence initial language token is required in the form of >>id<< (id = valid target language ID)
```

or by using `langId` parameter if all your dataset has 1 single language.

```python
# this if for instance everything is German
marian = MarianTransformer.pretrained("opus_mt_mul_en", "xx")\
 .setInputCols(["sentences"])\
 .setOutputCol("translation")\
 .setLangId("deu")

# or this if you have different languages inside

sdf = spark.createDataFrame([[">>deu<< Hallo wie geht es dir Ich bin hubert aus Deutschland"], 
                             [">>fra<< Wikipédia est un projet d'encyclopédie collective en ligne, universelle, multilingue et fonctionnant sur le principe du wiki. Ce projet vise à offrir un contenu librement réutilisable, objectif et vérifiable, que chacun peut modifier et améliorer."]]).toDF("text") 

```

These are available `>>id<<`:

```
source language(s): abk acm ady afb afh_Latn afr akl_Latn aln amh ang_Latn apc ara arg arq ary arz asm ast avk_Latn awa aze_Latn bak bam_Latn bel bel_Latn ben bho bod bos_Latn bre brx brx_Latn bul bul_Latn cat ceb ces cha che chr chv cjy_Hans cjy_Hant cmn cmn_Hans cmn_Hant cor cos crh crh_Latn csb_Latn cym dan deu dsb dtp dws_Latn egl ell enm_Latn epo est eus ewe ext fao fij fin fkv_Latn fra frm_Latn frr fry fuc fuv gan gcf_Latn gil gla gle glg glv gom gos got_Goth grc_Grek grn gsw guj hat hau_Latn haw heb hif_Latn hil hin hnj_Latn hoc hoc_Latn hrv hsb hun hye iba ibo ido ido_Latn ike_Latn ile_Latn ilo ina_Latn ind isl ita izh jav jav_Java jbo jbo_Cyrl jbo_Latn jdt_Cyrl jpn kab kal kan kat kaz_Cyrl kaz_Latn kek_Latn kha khm khm_Latn kin kir_Cyrl kjh kpv krl ksh kum kur_Arab kur_Latn lad lad_Latn lao lat_Latn lav ldn_Latn lfn_Cyrl lfn_Latn lij lin lit liv_Latn lkt lld_Latn lmo ltg ltz lug lzh lzh_Hans mad mah mai mal mar max_Latn mdf mfe mhr mic min mkd mlg mlt mnw moh mon mri mwl mww mya myv nan nau nav nds niu nld nno nob nob_Hebr nog non_Latn nov_Latn npi nya oci ori orv_Cyrl oss ota_Arab ota_Latn pag pan_Guru pap pau pdc pes pes_Latn pes_Thaa pms pnb pol por ppl_Latn prg_Latn pus quc qya qya_Latn rap rif_Latn roh rom ron rue run rus sag sah san_Deva scn sco sgs shs_Latn shy_Latn sin sjn_Latn slv sma sme smo sna snd_Arab som spa sqi srp_Cyrl srp_Latn stq sun swe swg swh tah tam tat tat_Arab tat_Latn tel tet tgk_Cyrl tha tir tlh_Latn tly_Latn tmw_Latn toi_Latn ton tpw_Latn tso tuk tuk_Latn tur tvl tyv tzl tzl_Latn udm uig_Arab uig_Cyrl ukr umb urd uzb_Cyrl uzb_Latn vec vie vie_Hani vol_Latn vro war wln wol wuu xal xho yid yor yue yue_Hans yue_Hant zho zho_Hans zho_Hant zlm_Latn zsm_Latn zul zza

target language(s): eng
```

At the bottom of the model, in the source, you can refer to the original model to see these language identifiers.
https://nlp.johnsnowlabs.com/2020/12/28/opus_mt_mul_en_xx.html 

That's being said, please leave this issue open and I am going to keep the `bug` label because I couldn't get translation despite providing either `langId` or inline language. (I do remember a regex issue, but I will dig deeper to see what went wrong)

**Comment by maziyarpanahi** [comment]
@C-K-Loan 

Unfortunately, the issue is actually with those two models. I am not sure if they are deprecated or you should report this issue on HuggingFace, but they lack language codes in their models:

- https://huggingface.co/Helsinki-NLP/opus-mt-mul-en/blob/main/tokenizer_config.json

An example of a model supporting multiple languages:

- https://huggingface.co/Helsinki-NLP/opus-mt-en-ROMANCE/blob/main/tokenizer_config.json

Could you please open an issue on Transformers reporting that the following two models don't have any supported languages?

```python
# opus-mt-mul-en
# opus-mt-en-mul

from transformers import MarianMTModel, MarianTokenizer

model_name = 'Helsinki-NLP/opus-mt-mul-en'
tokenizer = MarianTokenizer.from_pretrained(model_name)

tokenizer.supported_language_codes # this returns nothing
```

**Comment by maziyarpanahi** [comment]
Fixed in Spark NLP 3.3.3 release
